### ISSUE TITLE
curl: brew-installed curl does not work

### ISSUE THREAD

**Issue by biocyberman** [Linuxbrew/linuxbrew#832]
My brew is clean reinstalled because there were errors `Error: Failed to download resource "xxx"`

In this new installion, found out that after brew install curl, it can't download resource any longer. So I think there is something not configured correctly for curl installation. If I uninstall curl, every thing works again. Worth to note that `brew unlink curl ` did not help.


  ```bash
% brew install git    # curl was installed by brew already                                                                                                                                                                                                                                                              !838
==> Downloading https://www.kernel.org/pub/software/scm/git/git-2.7.1.tar.xz
Error: Failed to download resource "git"
Download failed: https://www.kernel.org/pub/software/scm/git/git-2.7.1.tar.xz

  % brew uninstall curl                                                                                                                                                                                                                                                               !840
Uninstalling ~/.linuxbrew/Cellar/curl/7.47.1... (359 files, 2.7M)

  % brew install git --ignore-dependencies                                                                                                                                                                                                                                            !841
==> Downloading https://www.kernel.org/pub/software/scm/git/git-2.7.1.tar.xz
######################################################################## 100.0%
==> make install prefix=~/.linuxbrew/Cellar/git/2.7.1 sysconfdir=~/.linuxbrew/etc CC=/usr/bin/gcc CFLAGS=-Os -w -pipe -march=core2 LDFLAGS=-L~/.linuxbrew/lib -Wl,--dynamic-linker=~/.linuxbrew/opt/glibc/lib/ld-linux-x86-64.so.2 -Wl,-rpath,~/.li
==> make CC=/usr/bin/gcc CFLAGS=-Os -w -pipe -march=core2 LDFLAGS=-L~/.linuxbrew/lib -Wl,--dynamic-linker=~/.linuxbrew/opt/glibc/lib/ld-linux-x86-64.so.2 -Wl,-rpath,~/.linuxbrew/lib
==> Downloading https://www.kernel.org/pub/software/scm/git/git-manpages-2.7.1.tar.xz
######################################################################## 100.0%
==> Downloading https://www.kernel.org/pub/software/scm/git/git-htmldocs-2.7.1.tar.xz
######################################################################## 100.0%
## trimmed for clarity ####
  % brew --config                                                                                                                                                                                                                                                                     !842
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Linuxbrew/linuxbrew.git
HEAD: 38c3087453a10567f228e7e2a472fc61beed9986
Last commit: 12 hours ago
HOMEBREW_PREFIX:~/.linuxbrew
HOMEBREW_REPOSITORY: ~/.linuxbrew
HOMEBREW_CELLAR: ~/.linuxbrew/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://linuxbrew.bintray.com
CPU: 8-core 64-bit nehalem
Kernel: Linux 2.6.32-573.12.1.el6.674g0000.x86_64 x86_64 GNU/Linux
OS: CentOS release 6.7 (Final)
OS glibc: 2.12
OS gcc: 4.4.7
Linuxbrew glibc: 2.19
Linuxbrew gcc: N/A
Clang: N/A
X11: 2.7.8 => /usr
System Ruby: 1.8.7-p374 => /usr/bin/ruby
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby
Java: 1.7.0_91

```

This is how brew install curl:

```bash
brew install curl                                                                                                                                                                                                                                                                 !843
==> Downloading https://curl.haxx.se/download/curl-7.47.1.tar.bz2
Already downloaded: /data/users/vql/.cache/Homebrew/curl-7.47.1.tar.bz2
==> ./configure --disable-silent-rules --prefix=~/.linuxbrew/Cellar/curl/7.47.1 --with-ssl=~/.linuxbrew/opt/openssl --with-ca-bundle=~/.linuxbrew/etc/openssl/cert.pem --without-libssh2 --without-libidn --without-libmetalink --without-gssapi --without-librtmp 
==> make install
==> Caveats
zsh completion has been installed to:
  ~/.linuxbrew/share/zsh/site-functions

==> Summary
~/.linuxbrew/Cellar/curl/7.47.1: 359 files, 2.7M, built in 33 seconds
```

**Comment by sjackman** [comment]
I believe this bug is a duplicate of #797

**Comment by sjackman** [comment]
There is a workaround.
```sh
brew fetch git
brew install git
```

**Comment by biocyberman** [comment]
Yes, the workaround works for me.

**Comment by sjackman** [comment]
I wish I knew what was going on here. So strange.

**Comment by maxim-belkin** [comment]
I can only confirm that in order to upgrade `openssl` I have to do the same thing:
```
brew fetch openssl
brew upgrade openssl
```
Is it possible to fix the behavior of `brew upgrade` to take these kind of problems into account. That is, when `curl` is linuxbrew's `curl`, then it for sure depends on `openssl`, and then upgrading either one should be done as "fetch then upgrade". I'm sure, but maybe "fetch then upgrade" might be a good idea in general...

**Comment by sjackman** [comment]
This issue was caused by a particular bug that affected both `install` and `upgrade` and was resolved by #841. The `upgrade` issue that you're seeing is different, but also quite real, and also affects upgrading the dependencies of the compiler and any other "core" formula. Please open a new issue to discuss.

**Comment by erigler-usgs** [comment]
I haven't seen a new issue opened by @maxim-belkin as requested by @sjackman re. the cURL/openssl upgrade catch22, something I recently ran into (if I missed it, apologies). I opened issue #883 to initiate discussion.
